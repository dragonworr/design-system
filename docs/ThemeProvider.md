
# ThemeProvider

The ThemeProvider component is a wrapper around the styled-components' [ThemeProvider][sc-theme]

Wrap the root of your application with the `ThemeProvider` component,
which adds the Design System theme to context for use in styled-components
and sets typographic defaults.
This should only be included once in your application.

```jsx
<ThemeProvider>
  <Heading>
    Hello
  </Heading>
</ThemeProvider>
```

Prop | Type | Description
---|---|---
`legacy` | Boolean | Enable legacy color palette
`customBreakpoints` | Array | Array of em number values to override default breakpoints

[sc-theme]: https://www.styled-components.com/docs/advanced#theming
