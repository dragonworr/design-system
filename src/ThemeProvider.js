import React from 'react'
import styled, {
  ThemeProvider as StyledThemeProvider
} from 'styled-components'
import nextTheme from './theme'
import legacyTheme from './legacyTheme'

export const Base = styled.div`
  font-family: ${props => props.theme.font};
  line-height: 1.4;

  * {
    box-sizing: border-box;
  }
`

const ThemeProvider = ({ legacy, ...props }) => {
  const theme = legacy ? legacyTheme : nextTheme

  return (
    <StyledThemeProvider theme={theme}>
      <Base {...props} />
    </StyledThemeProvider>
  )
}

export default ThemeProvider
