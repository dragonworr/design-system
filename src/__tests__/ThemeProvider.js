import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { ThemeProvider, theme } from '..'
import { Base } from '../ThemeProvider'

describe('ThemeProvider', () => {
  test('renders', () => {
    const json = renderer.create(<ThemeProvider />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('Base component includes a font', () => {
    const json = renderer.create(<Base />).toJSON()
    expect(json).toMatchSnapshot()
    expect(json).toHaveStyleRule('font-family', theme.font)
  })
})

