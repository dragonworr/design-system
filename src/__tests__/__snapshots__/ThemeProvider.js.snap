// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider Base component includes a font 1`] = `
.c0 {
  font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif;
}

<div
  className="c0"
/>
`;

exports[`ThemeProvider renders 1`] = `
.c0 {
  font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif;
}

<div
  className="c0"
/>
`;

exports[`ThemeProvider renders with legacy prop 1`] = `
.c0 {
  font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif;
}

<div
  className="c0"
/>
`;
